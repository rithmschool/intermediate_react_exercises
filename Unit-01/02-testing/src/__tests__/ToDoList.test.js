import React from "react";
import ToDoList from "../ToDoList";
import NewToDoForm from "../NewToDoForm";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import Enzyme, { shallow, mount, render } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("<ToDoList />", () => {
	it("matches the snapshot", () => {
		var tree = renderer.create(<ToDoList />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("renders the NewToDoForm component", () => {
		const wrapper = shallow(<ToDoList />);
		expect(wrapper.find(NewToDoForm)).toHaveLength(1);
	});

	test("should detect state and all 3 default todos on mounting", () => {
		const wrapper = mount(<ToDoList />);
		expect(wrapper.state("toDos")).toHaveLength(3);
		expect(wrapper.state("keyNum")).toBe(3);
	});

	test("should have 4 todos upon adding", () => {
		const wrapper = mount(<ToDoList />);
		wrapper.instance().handleSubmit("test");
		expect(wrapper.state("toDos")).toHaveLength(4);
		expect(wrapper.state("keyNum")).toBe(4);
	});
});
