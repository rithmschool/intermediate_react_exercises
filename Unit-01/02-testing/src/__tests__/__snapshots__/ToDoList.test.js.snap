// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDoList /> matches the snapshot 1`] = `
<main>
  <p>
    <a
      onClick={[Function]}
    >
       
      <img
        alt="Unfinished"
        src="https://freeclipartspot.com//storage/upload/square-clip-art/square-clip-art-6.png"
      />
       
    </a>
    <span
      className="ToDo"
    >
       
      Finish Homework
       
    </span>
    <span
      className="ButtonSection"
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "yellow",
          }
        }
      >
        Edit
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      >
        Delete
      </button>
    </span>
  </p>
  <p>
    <a
      onClick={[Function]}
    >
       
      <img
        alt="Unfinished"
        src="https://freeclipartspot.com//storage/upload/square-clip-art/square-clip-art-6.png"
      />
       
    </a>
    <span
      className="ToDo"
    >
       
      Sleep
       
    </span>
    <span
      className="ButtonSection"
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "yellow",
          }
        }
      >
        Edit
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      >
        Delete
      </button>
    </span>
  </p>
  <p>
    <a
      onClick={[Function]}
    >
       
      <img
        alt="Finished"
        src="http://is2.mzstatic.com/image/thumb/Purple5/v4/0e/52/a8/0e52a81b-bbe0-69fe-9d86-a3a47d5a9c63/source/1200x630bb.jpg"
      />
       
    </a>
    <span
      className="ToDo"
    >
       
      Be Happy
       
    </span>
    <span
      className="ButtonSection"
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "yellow",
          }
        }
      >
        Edit
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      >
        Delete
      </button>
    </span>
  </p>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="toDo"
    >
      To do:
    </label>
    <input
      id="toDo"
      name="toDo"
      onChange={[Function]}
      placeholder="What do you need to do?"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add to-do item!
    </button>
  </form>
</main>
`;
